api/v1beta1/awscluster_types.go

```diff
@@ -56,33 +56,46 @@ type AWSClusterSpec struct {
 	// +optional
 	ControlPlaneLoadBalancer *AWSLoadBalancerSpec `json:"controlPlaneLoadBalancer,omitempty"`
 
-	// ImageLookupFormat is the AMI naming format to look up machine images when
-	// a machine does not specify an AMI. When set, this will be used for all
-	// cluster machines unless a machine specifies a different ImageLookupOrg.
-	// Supports substitutions for {{.BaseOS}} and {{.K8sVersion}} with the base
-	// OS and kubernetes version, respectively. The BaseOS will be the value in
-	// ImageLookupBaseOS or ubuntu (the default), and the kubernetes version as
-	// defined by the packages produced by kubernetes/release without v as a
-	// prefix: 1.13.0, 1.12.5-mybuild.1, or 1.17.3. For example, the default
+	// AMI ID is the first method for finding an AMI ID, if this field is set all other methods will be ignored.
+	// example: ami-1234567890abcdef0
+	AMI AMIReference `json:"ami,omitempty"`
```

Comment:
With these additional image fields it feels like we should refactor the AWSCluster and group these into a separate "image" struct? We'd need to do this as part of the v1beta2 API version bump.

Reply:
It can then be used by AWSMCP and AWSMachine as well.

Reply:
i agree, are we at a point to go to v1beta2? i'd love to do that full refactor

Reply:
maybe we just deprecate the stuff we dont care about and add all these fields to the AMIReference. I found this PR: https://github.com/kubernetes-sigs/cluster-api-provider-aws/commit/1956af1cb5fbfaf8469c5c215181e6acf5551f69 where we actually used to have filters and other things. it kind of feels like AMI Reference was your Image struct at one point...

Reply:
yup, let's do that. ill close this in favor of a new PR with what we discussed